flax/linen/fp8_ops.py

```diff
@@ -411,122 +349,142 @@ def quantized_dot_fwd(
-  preferred_element_type,
-  res,
-  g
+    dimension_numbers,
+    preferred_element_type,
+    res,
+    g
 ):
   (
-    lhs,
-    q_lhs,
-    lhs_scale,
-    rhs,
-    q_rhs,
-    rhs_scale,
-    out_grad_scale,
-    out_grad_amax_history,
+      lhs,
+      q_lhs,
+      lhs_scale,
+      rhs,
+      q_rhs,
+      rhs_scale,
+      out_grad_scale,
+      out_grad_amax_history,
   ) = res
 
   new_out_grad_scale, new_out_grad_amax_history = update_fp8_meta(
-    g,
-    jnp.float8_e5m2,
-    out_grad_scale,
-    out_grad_amax_history,
+      g,
+      jnp.float8_e5m2,
+      out_grad_scale,
+      out_grad_amax_history,
   )
 
-  q_g = quantize(g, jnp.float8_e5m2, _fm32_to_float32(new_out_grad_scale), preferred_element_type)
+  q_g = quantize(
+      g,
+      jnp.float8_e5m2,
+      _fm32_to_float32(new_out_grad_scale),
+      preferred_element_type
+  )
 
   grad_lhs = dot_general_transpose_lhs(
-    q_g,
-    lhs,
-    q_rhs,
-    dimension_numbers=dimension_numbers,
-    precision=lax.Precision.HIGHEST,
-    preferred_element_type=preferred_element_type,
+      q_g,
+      lhs,
+      q_rhs,
+      dimension_numbers=dimension_numbers,
+      precision=lax.Precision.HIGHEST,
+      preferred_element_type=preferred_element_type,
   )
+
   grad_lhs = dequantize(
-    grad_lhs,
-    preferred_element_type,
-    _fm32_to_float32(rhs_scale) * _fm32_to_float32(new_out_grad_scale)
+      grad_lhs,
+      preferred_element_type,
+      _fm32_to_float32(rhs_scale) * _fm32_to_float32(new_out_grad_scale)
   )
 
   grad_rhs = dot_general_transpose_rhs(
-    q_g,
-    q_lhs,
-    rhs,
-    dimension_numbers=dimension_numbers,
-    precision=lax.Precision.HIGHEST,
-    preferred_element_type=preferred_element_type,
+      q_g,
+      q_lhs,
+      rhs,
+      dimension_numbers=dimension_numbers,
+      precision=lax.Precision.HIGHEST,
+      preferred_element_type=preferred_element_type,
   )
+
   grad_rhs = dequantize(
-    grad_rhs,
-    preferred_element_type,
-    _fm32_to_float32(lhs_scale) * _fm32_to_float32(new_out_grad_scale)
+      grad_rhs,
+      preferred_element_type,
+      _fm32_to_float32(lhs_scale) * _fm32_to_float32(new_out_grad_scale)
   )
 
   return (
-    grad_lhs,
-    None,
-    None,
-    grad_rhs,
-    None,
-    None,
-    new_out_grad_scale,
-    new_out_grad_amax_history,
+      grad_lhs,
+      None,
+      None,
+      grad_rhs,
+      None,
+      None,
+      new_out_grad_scale,
+      new_out_grad_amax_history,
   )
 
 quantized_dot.defvjp(quantized_dot_fwd, quantized_dot_bwd)
 
-# Convenience wrappers for the quantize-dot-dequantize
-def q_dot_dq(
-  lhs,
-  rhs,
-  lhs_scale,
-  rhs_scale,
-  out_grad_scale,
-  lhs_amax_history,
-  rhs_amax_history,
-  out_grad_amax_history,
-  compute_dtype,
-  dimension_numbers,
-  precision=None,
-  preferred_element_type=None
+# Wrapper function to achieve the same effect as the dot_general function
+# but with fp8 quantization and dequantization.
+def fp8_scaled_dot_general(
```

Comment:
The lax.dot_general is defined that dimension_numbers, precision and preferred_element_type are NOT kw_args. Please consider move preferred_element_type to non-kw args.

Reply:
Done.